ompi/mca/pml/ob1/pml_ob1.c

```diff
@@ -264,15 +268,17 @@ int mca_pml_ob1_add_comm(ompi_communicator_t* comm)
              * situation as the cant_match is only checked when a new fragment is received from
              * the network.
              */
+            SPC_TIMER_START(OMPI_OOS_MATCH_TIME, &timer);
             OPAL_LIST_FOREACH(frag, &pml_proc->frags_cant_match, mca_pml_ob1_recv_frag_t) {
-               hdr = &frag->hdr.hdr_match;
-               /* If the message has the next expected seq from that proc...  */
-               if(hdr->hdr_seq != pml_proc->expected_sequence)
-                   continue;
-
-               opal_list_remove_item(&pml_proc->frags_cant_match, (opal_list_item_t*)frag);
-               goto add_fragment_to_unexpected;
-           }
+                hdr = &frag->hdr.hdr_match;
+                /* If the message has the next expected seq from that proc...  */
+                if(hdr->hdr_seq != pml_proc->expected_sequence)
+                    continue;
+                
+                opal_list_remove_item(&pml_proc->frags_cant_match, (opal_list_item_t*)frag);
+                goto add_fragment_to_unexpected;
```

Comment:
Do you fail to account the matching time when the goto is taken ?

Reply:
Yes, good catch.  I moved the start timer to before the add_fragment_to_unexpected: and the stop timer to after the end of the else bracket.